src/librados/RadosClient.cc

```diff
@@ -612,17 +601,13 @@ int librados::RadosClient::wait_for_osdmap()
 
 int librados::RadosClient::wait_for_latest_osdmap()
 {
-  Mutex mylock("RadosClient::wait_for_latest_osdmap");
-  Cond cond;
-  bool done;
-
-  objecter->wait_for_latest_osdmap(new C_SafeCond(&mylock, &cond, &done));
-
-  mylock.Lock();
-  while (!done)
-    cond.Wait(mylock);
-  mylock.Unlock();
-
+  auto f = objecter->wait_for_latest_osdmap(boost::asio::use_future);
+  f.wait();
+  try {
+    f.get();
+  } catch (const boost::system::system_error& e) {
+    return ceph::from_error_code(e.code());
+  }
   return 0;
```

Comment:
Should this be replaced with some kind of "success" mnemonic?

Reply:
This function is still returning ints for compatibility, though we can move the boundary later.